SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/UnresolvedReferenceGymTest/UnresolvedReferenceGymTest.cpp

```diff
@@ -0,0 +1,57 @@
+
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "UnresolvedReferenceGymTest.h"
+
+AUnresolvedReferenceGymTest::AUnresolvedReferenceGymTest()
+	: Super()
+{
+	Author = "Ajanth";
+	Description = TEXT("Tests what happens when structs with references to actors whose entity have not been created yet are replicated.");
+}
+
+void AUnresolvedReferenceGymTest::PrepareTest()
+{
+	Super::PrepareTest();
+
+	AddStep(TEXT("MakeActorReferences"), FWorkerDefinition::Server(1), nullptr, [this]() {
+		AUnresolvedReferenceGymTestActor* ActorRefsActor =
+			GetWorld()->SpawnActor<AUnresolvedReferenceGymTestActor>({ 0.0f, 0.0f, 0.0f }, FRotator::ZeroRotator);
+		RegisterAutoDestroyActor(ActorRefsActor);
+
+		for (int i = 0; i < 3; i++)
+		{
+			AReplicatedTestActorBase* Actor =
+				GetWorld()->SpawnActor<AReplicatedTestActorBase>(FVector(0, 0, 0), FRotator::ZeroRotator, FActorSpawnParameters());
+			(ActorRefsActor->ActorRefs).Add(Actor);
+			RegisterAutoDestroyActor(Actor);
+		}
+
+		FinishStep();
+	});
+
+	AddStep(TEXT("CheckNoNullReferences"), FWorkerDefinition::AllClients, nullptr, nullptr, [this](float deltaTime) {
+		for (TActorIterator<AUnresolvedReferenceGymTestActor> Iter(GetWorld()); Iter; ++Iter)
```

Comment:
```suggestion
		for (const AUnresolvedReferenceGymTestActor* Actor: TActorRange<AUnresolvedReferenceGymTestActor>(GetWorld()))```
Minor: TActor range has nicer semantics.